### PR TITLE
[DPE-5289]: CI uses new juju versions

### PR DESCRIPTION
**Pull request by Gu1nness**
## Issue

We need to keep our CI up to date.

## Solution

* Use juju 3.5.3 in regular runs
* Runs also 3.6/beta on nightly runs

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -55,10 +55,10 @@ juju = "^3.4.0"
 pytest = "^8.1.1"
```
Comment (Mehdi-Bendriss) [inline pyproject.toml:56]: above this line, could you change the pylibjuju version to `3.5.0.0`?

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -67,17 +67,26 @@ jobs:
...
 
   integration-test:
-    name: Integration test charm | 3.4.4
+    strategy:
+      fail-fast: false
+      matrix:
+        juju:
+          - agent: 3.5.3 # renovate: juju-agent-pin-minor
+            allure_report: true
+          - snap_channel: 3.6/beta
```
Comment (Mehdi-Bendriss) [inline .github/workflows/ci.yaml]: 3.5/stable ?

**Review by Mehdi-Bendriss — CHANGES_REQUESTED** [review]

**Review by carlcsaposs-canonical — APPROVED** [review]
you may need to specify base when deploying with libjuju so that it doesn't default to focal cc @shayancanonical

**Review by Mehdi-Bendriss — APPROVED** [review]

**Review by MiaAltieri — APPROVED** [review]
LGTM - don't worry about the TLS tests, those are flakey and an issue has been filed

**Pull request merged**